forte/pipeline.py

```diff
@@ -466,6 +471,43 @@ def save(self, path: str):
         with open(path, "w", encoding="utf-8") as f:
             yaml.safe_dump(self._dump_to_config(), f)
 
+    def export(self, name: Optional[str] = None) -> Optional[str]:
+        r"""Export pipeline to FORTE_EXPORT_PATH
+        FORTE_EXPORT_PATH is an environment variable
+        The pipeline is saved by :meth:`~forte.pipeline.Pipeline.save`
+        This method will have the following behaviors:
+        - FORTE_EXPORT_PATH will be created if assigned but not found.
+        - If name is not provided, a default one will be used.
+        - When the pipeline is exported multiple time with default name,
+        the export name will be `pipeline-1.yml`, `pipeline-2.yml`...
+        - Raise exception if provided name is taken.
+
+        Args:
+            name: Export name of the pipeline. Default is `pipeline`.
+
+        Returns:
+            Optional[str]: Export path of pipeline config YAML.
+
+        Raises:
+            ValueError: if export name is already taken.
+        """
+        export_path: Optional[str] = os.environ.get("FORTE_EXPORT_PATH")
+        if export_path:
+            os.makedirs(export_path, exist_ok=True)
+
+            if name:
+                export_name = f"{name}.yml"
+            else:
+                export_name = f"pipeline-{self._export_count}.yml"
+                self._export_count += 1
```

Comment:
Actually, this implementation will likely fail. We should check the duplicate file in the directory and then increment the suffix if necessary. Storing a state will cause several problems, the main one is this value will be lost when the pipeline is rerun.

Reply:
So instead using count as suffix, how about using timestamp like `2022.03.18.07.39.27`?